x-pack/plugins/enterprise_search/public/applications/shared/result/result_title.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SearchHit } from '@elastic/elasticsearch/lib/api/types';
+
+const TITLE_KEYS = ['title', 'name'];
+
+const isRecord = (source: unknown | undefined): source is Record<PropertyKey, unknown> => {
+  return source !== undefined && typeof source === 'object' && source !== null;
+};
+
+function hasStringKey<T extends Record<PropertyKey, unknown>, K extends PropertyKey>(
+  source: T,
+  key: K
+): source is T & Record<K, string> {
+  return source.hasOwnProperty(key) && typeof source[key] === 'string';
```

Comment:
I'm wondering whether the source.hasOwnProperty[key] is really necessary here. This is always an object constructed from a JSON blob, there's no case in which source[key] is going to be a string but not an own property, right?

Reply:
I think you're right, I'll double check and remove it. I tend to write overly defensive explicit code on the first pass.